.integration/tests/metrics/mod.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::wait_until;
+
+use snarkos_integration::{ClientNode, TestNode};
+use snarkos_metrics as metrics;
+
+use pea2pea::Pea2Pea;
+
+#[tokio::test]
+async fn metrics_initialization() {
+    // Start a test node.
+    let _test_node = TestNode::default().await;
+
+    // Initialise the metrics, we need to call this manually in tests.
+    let snapshotter = metrics::initialize();
+
+    // Start a snarkOS node.
+    let _client_node = ClientNode::default().await;
+
+    // Verify the metrics have been properly initialised, expect the block height to be set.
+    assert_eq!(
+        metrics::get_metric(&snapshotter, metrics::blocks::HEIGHT),
+        metrics::MetricVal::Gauge(0.0)
+    );
+    assert_eq!(
+        metrics::get_metric(&snapshotter, metrics::peers::CONNECTED),
+        metrics::MetricVal::Gauge(0.0)
+    );
+    assert_eq!(
+        metrics::get_metric(&snapshotter, metrics::peers::CANDIDATE),
+        metrics::MetricVal::Gauge(0.0)
+    );
+    assert_eq!(
+        metrics::get_metric(&snapshotter, metrics::peers::RESTRICTED),
+        metrics::MetricVal::Gauge(0.0)
+    );
+
+    // Clear the recorder to avoid the global state bleeding into other tests.
+    metrics::clear_recorder();
```

Comment:
if all the metrics tests begin with setting up a `Snapshotter` and end with a call to `metrics::clear_recorder()`, perhaps we could make our own wrapper `struct` for the former that would call the latter in a custom `Drop` impl?